Templating/TemplateGuesser.php

```diff
@@ -30,13 +30,22 @@ class TemplateGuesser
     protected $kernel;
 
     /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel             A KernelInterface instance
+     * @param string[]        $controllerPatterns Regexps extracting the controller name from its FQN.
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, array $controllerPatterns = array())
     {
+        $controllerPatterns[] = '/Controller\\\(.+)Controller$/';
```

Comment:
The configuration you show was exactly what I've in mind (it should be the default one).
Most people will never change the default config but some experimented will.

The main benefit is: "I want `@Template` support for new Actions but not for my Legacy controller".

But it's a detail, I personally never use `@Template` so I'll not fight for it :)